scripts/needs_ci_run

```diff
@@ -35,14 +31,24 @@ def last_build_successful():
     token = environ.get('CIRCLE_API_TOKEN')
 
     if not branch:
-        print('CIRCLE_BRANCH env not set! Unable to do CI skip check')
-        sys.exit(0)
+        print('CIRCLE_BRANCH env not set! Unable to do CI skip check', file=sys.stderr)
+        print('needs CI')
+        return False, None
     if not current_workflow_id:
-        print('CIRCLE_WORKFLOW_ID env not set! Unable to do CI skip check')
-        sys.exit(0)
+        print('CIRCLE_WORKFLOW_ID env not set! Unable to do CI skip check', file=sys.stderr)
+        print('needs CI')
+        return False, None
     if not token:
-        print('CIRCLE_API_TOKEN env not set! Unable to do CI skip check')
-        sys.exit(0)
+        print('CIRCLE_API_TOKEN env not set! Unable to do CI skip check', file=sys.stderr)
+        print('needs CI')
+        return False, None
+
+    # If master, we run 4 jobs, if PR, we run 3 (no docker workflow)
+    # jobs defined in .circleci/config.yml
+    if branch == "master":
+        JOBS = ['build-docker-images', 'lint-build-debug', 'build-release', 'build-osx']
+    else:
+        JOBS = ['lint-build-debug', 'build-release', 'build-osx']
```

Comment:
sorry to barge in here like that, but I realized it's _still_ not skipping CI and seems the reason is that in #3683 we (finally) disabled the docker workflow for PRs but didn't update this file to reflect that, so it was always failing when it didn't get the `build-docker-images` workflow ID for PRs.

Also added some early `sys.exit` when relevant

Reply:
oops just messed that up, didnt realize you added that, apologies!